src/openzaak/utils/tasks.py

```diff
@@ -0,0 +1,791 @@
+            error_message = (
+                f"UUID given on row {row_index} was already found! Not overwriting "
+                "existing EIO."
+            )
+
+            logger.warning(error_message)
+            document_row.comment = error_message
+            document_row.processed = True
+
+            return document_row
+
+        data["uuid"] = str(uuid)
+    else:
+        data["uuid"] = str(uuid4())
+
+    gegevensgroep_fields = ("ondertekening", "integriteit")
+
+    for field in gegevensgroep_fields:
+        gegevens_groep_value = data.pop(field, {})
+
+        for key, value in gegevens_groep_value.items():
+            data[f"{field}_{key}"] = value
+
+    if "vertrouwelijkheidaanduiding" not in data:
+        informatieobjecttype = data["informatieobjecttype"]
+        data[
+            "vertrouwelijkheidaanduiding"
+        ] = informatieobjecttype.vertrouwelijkheidaanduiding
+
+    instance = EnkelvoudigInformatieObject(**data)
+
+    if not instance.identificatie:
+        instance.identificatie = generate_unique_identification(
+            instance, "creatiedatum"
+        )
+
+    try:
+        instance.clean()
+    except ValidationError as e:
+        error_message = (
+            "A validation error occurred while validating a "
+            f"EnkelvoudigInformtatieObject on line {row_index}: \n"
+            f"{str(e)}"
+        )
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    file_path = document_row.bestandspad
+    path = Path(settings.IMPORT_DOCUMENTEN_BASE_DIR) / Path(file_path)
+
+    if not path.exists() or not path.is_file():
+        error_message = (
+            f"The given filepath {path} does not exist or is a file for "
+            f"row {row_index}"
+        )
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    default_dir = _get_default_path(instance.inhoud)
+    import_path = f"{default_dir}/{path.name}"
+
+    if not default_dir.exists():
+        default_dir.mkdir(parents=True)
+
+    try:
+        shutil.copy2(path, import_path)
+    except Exception as e:
+        error_message = f"Unable to copy file for row {row_index}: \n {str(e)}"
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    instance.inhoud.name = str(import_path)
+
+    document_row.instance = instance
+    return document_row
+
+
+@transaction.atomic()
+def _batch_create_eios(batch: list[DocumentRow], zaak_uuids: dict[str, int]) -> None:
+    canonicals = []
+
+    for row in batch:
+        if row.instance is None:
+            continue
+
+        canonical = EnkelvoudigInformatieObjectCanonical()
+        row.instance.canonical = canonical
+
+        canonicals.append(canonical)
```

Comment:
Can't this be done as part of `_import_document_row`, to avoid doing the same loop multiple times?